ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayloadV3.java

```diff
@@ -47,19 +54,24 @@ public String getName() {
   }
 
   @Override
-  protected ValidationResult<RpcErrorType> validateForkSupported(
-      final Object reqId, final EnginePayloadParameter payloadParameter) {
-    var cancun = timestampSchedule.hardforkFor(s -> s.fork().name().equalsIgnoreCase("Cancun"));
+  protected ValidationResult<RpcErrorType> validateParamsAndForkSupported(
+      final Object reqId,
+      final EnginePayloadParameter payloadParameter,
+      final Optional<List<String>> maybeVersionedHashParam,
+      final Optional<Bytes32> maybeParentBeaconBlockRoot) {
 
-    if (cancun.isPresent() && payloadParameter.getTimestamp() >= cancun.get().milestone()) {
-      if (payloadParameter.getBlobGasUsed() == null
-          || payloadParameter.getExcessBlobGas() == null) {
-        return ValidationResult.invalid(RpcErrorType.INVALID_PARAMS, "Missing blob gas fields");
-      } else {
-        return ValidationResult.valid();
-      }
-    } else {
+    if (payloadParameter.getBlobGasUsed() == null || payloadParameter.getExcessBlobGas() == null) {
+      return ValidationResult.invalid(RpcErrorType.INVALID_PARAMS, "Missing blob gas fields");
+    } else if (maybeVersionedHashParam == null) {
+      return ValidationResult.invalid(
+          RpcErrorType.INVALID_PARAMS, "Missing versioned hashes field");
+    } else if (maybeParentBeaconBlockRoot.isEmpty()) {
+      return ValidationResult.invalid(
+          RpcErrorType.INVALID_PARAMS, "Missing parent beacon block root field");
+    }
+    if (payloadParameter.getTimestamp() < cancunTimestamp) {
```

Comment:
per https://github.com/hyperledger/besu/issues/5738 this check should be moved above the checking for null fields

Reply:
Changing this fixed a few hive tests